### ISSUE TITLE
WASM validation fails for compressed runtime blob

### ISSUE THREAD

**Issue by mnaamani** [Joystream/pioneer#4122]
When building the runtime WASM blobs produced are as follows:

```
-rw-r--r--  1 mokhtar  staff   1.0M Jan 23 11:41 joystream_node_runtime.compact.compressed.wasm
-rw-r--r--  1 mokhtar  staff   5.0M Jan 23 11:41 joystream_node_runtime.compact.wasm
-rwxr-xr-x  1 mokhtar  staff   5.6M Jan 23 11:41 joystream_node_runtime.wasm
```

We use the compressed version because the size of the non-compressed blobs are too large and do not fit in a block. The runtime also limits the size of the blob to ~3MB for proposals.

Pioneer fails to validate when using the compressed version.

![runtime-upgrade-proposal](https://user-images.githubusercontent.com/1621012/214811582-934e2e7a-1860-4249-9893-3ff19a971322.png)

Validation is performed here when processing the dropped file:
https://github.com/Joystream/pioneer/blob/9d4e04c3a0c945f6c1774475a0a57c8a031c6d0f/packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/RuntimeUpgrade.tsx#L45

We need a step to decompress the file first before validating it, but the compressed file is the final one that should actually be used in the proposal details.

The compression algorith used is: https://github.com/facebook/zstd
Reference in substrate node where it is using `zstd` cargo crate https://crates.io/crates/zstd

Will need to find an implementation in javascript to use in pioneer. Here is one but I have not tested it.
https://www.npmjs.com/package/zstandard-wasm

**Comment by dmtrjsg** [comment]
This one seems trivial, once ephesus Pioneer scope is done, Theo could take this one on to turnaround in a few hours.

cc @mnaamani @bwhm

**Comment by mnaamani** [comment]
Reference implementation to handle decompressing of wasm blob and inspecting runtime version

https://github.com/Joystream/joystream/pull/4629/files#diff-0b8c337c90045d20004df9d58183118f424b6bc70383c5f22fcde8ef0b90ea3f

**Comment by ivanturlakov** [comment]
Tested on https://pioneer-2-kvulka3th-joystream.vercel.app/#/proposals/current
wss://54.205.24.16.nip.io/ws-rpc

Used this blob https://pioneerapp.xyz/#/proposals/preview/175

⚠️ Validation was successful, but the button is inactive
<img width="1440" alt="Screenshot 2023-06-05 at 17 59 46" src="https://github.com/Joystream/pioneer/assets/46903215/6e6ff76a-9bf4-4a23-a11a-90a80b409029">

**Comment by thesan** [comment]
> :warning: Validation was successful, but the button is inactive

@ivanturlakov thank you. I'll remove this "fix" from the change log and release `1.5.0` today :rocket:

**Comment by dmtrjsg** [comment]
@chrlschwb we'd need community to help us with it, there's only one last thing remains
[in this comment](https://github.com/Joystream/pioneer/issues/4122#issuecomment-1576978647)
